dspace-api/src/main/java/org/dspace/eperson/GroupServiceImpl.java

```diff
@@ -144,7 +152,20 @@ public void addMember(Context context, Group groupParent, Group groupChild) thro
     }
 
     @Override
-    public void removeMember(Context context, Group group, EPerson ePerson) {
+    public void removeMember(Context context, Group group, EPerson ePerson) throws SQLException {
+        List<CollectionRole> collectionRoles = collectionRoleService.findByGroup(context, group);
+        if (!collectionRoles.isEmpty()) {
+            List<PoolTask> poolTasks = poolTaskService.findByGroup(context, group);
+            if (!poolTasks.isEmpty()) {
+                List<EPerson> ePeople = allMembers(context, group);
+                if (ePeople.size() == 1 && ePeople.contains(ePerson)) {
+                    throw new IllegalStateException(
+                            "Refused to remove user " + ePerson
+                                    .getID() + " from workflow group because the group " + group
+                                    .getID() + " has no other members");
```

Comment:
Now that you've added more details to the description, I see that this method is likely covering the scenario where the EPerson is the only member of a Workflow group with items assigned.

If that's the case, we probably should note that in the exception.  So, update the exception to say something like `"Refused to remove user [id] from workflow group because the group [id] has tasks assigned and no other members"`.

This same small update to the error message should also occur in the other `removeMember()` method below.

Reply:
This has been adjusted